### ISSUE TITLE
XMLHttpRequest Exception 101

### ISSUE THREAD

**Issue by bhalperin** [Telerik-Verified-Plugins/WKWebView#181]
This error popped up in my Cordova app on iPhone with iOS9, when the app sends a request to remote server (but only to one of 3 supposedly identical servers, selected based on user's identity, same request to the 2 other servers was fine).

No problems on iOS < 9. No idea what's wrong with Apple's UIWebView in iOS9. **All I know that once replaced with your WKWebView, above error simply vanished!** Go figure.

I'll need to test my app e2e to check that WKWebView has no negative effects elsewhere.

Thanks.

**Comment by adyz** [comment]
Yes, I face a similar issue. On my ios simulator 9.1 everything works fine, but on the real device with the same os, I get `XMLHttpRequest cannot load http://xxxx. Origin http://localhost:12344 is not allowed by Access-Control-Allow-Origin.

**Comment by EddyVerbruggen** [comment]
@adyz sounds like you need to add CORS headers to your http://xxxx server.

**Comment by adyz** [comment]
@EddyVerbruggen  - I did this since it work on the simulator, and also without this plugin works on the real device. Might be because of the port? Should I do something with my ajax requests? I've added `crossDomain: true` to my ajax setup.

**Comment by EddyVerbruggen** [comment]
WRT WKWebView: Simulator != Device. If you're seeing 'Origin http://localhost:12344 is not allowed by Access-Control-Allow-Origin' then the ONLY solution is to add CORS headers on the server.

If you don't control the server you need a proxy server in between which adds CORS headers and proxies to the upstream server.

**Comment by adyz** [comment]
I have this enabled on the server because I test my JS without cordova on http://localhost:3000 everyday...

Thanks for your help, I will investigate further on the server side too.

**Comment by adyz** [comment]
Also, on another real device with iOS 8.3 the build with this plugin works just fine.
I added `<key>NSAppTransportSecurity</key>` for all my request domains
